tests/force_insert_update/models.py

```diff
@@ -26,3 +26,26 @@ class SubCounter(Counter):
 class WithCustomPK(models.Model):
     name = models.IntegerField(primary_key=True)
     value = models.IntegerField()
+
+
+class GrandParent(models.Model):
+    name = models.CharField(max_length=50, null=True)
+    value = models.IntegerField(null=True)
+
+
+class Parent(GrandParent):
+    pass
+
+
+class Child(Parent):
+    pass
+
+
+class DiamondParent(GrandParent):
+    gp_ptr = models.OneToOneField(
+        GrandParent, primary_key=True, parent_link=True, on_delete=models.CASCADE
+    )
+
+
+class DiamondChild(Parent, DiamondParent):
+    pass
```

Comment:
Please try to reuse existing models (e.g. `Counter`, `SubCounter`) and minimize the number of new once.